src/ipa_pc/data_structures.rs

```diff
@@ -216,27 +278,83 @@ pub struct Proof<G: AffineCurve> {
     pub rand: Option<G::ScalarField>,
 }
 
-impl<G: AffineCurve> PCProof for Proof<G> {
-    fn size_in_bytes(&self) -> usize {
-        to_bytes![self].unwrap().len()
+impl<G: AffineCurve> SemanticallyValid for Proof<G> {
+    fn is_valid(&self) -> bool {
+        self.l_vec.is_valid() &&
+            self.r_vec.is_valid() &&
+            self.final_comm_key.is_valid() &&
+            self.c.is_valid() &&
+            if self.hiding_comm.is_some() { self.hiding_comm.as_ref().unwrap().is_valid() } else { true } &&
+            if self.rand.is_some() { self.rand.as_ref().unwrap().is_valid() } else { true }
```

Comment:
We can do some further consistency checks here: is the length of l_vec that of r_vec (if not tested before),  and does the length correspond the log len of the committer key of the PC scheme.

Reply:
I can do the first check you mentioned. Regarding the second one I can't, cause I don't have access to the committer key